packages/dashboard/src/components/schedule-visualizer/tests/waypoints-overlay.test.tsx

```diff
@@ -13,14 +12,21 @@ test('Render waypoints correctly', async () => {
   const nav_graphs = buildingMap.levels.flatMap((x) => x.nav_graphs);
   const waypoints = nav_graphs[0].vertices;
 
-  const wrapper = mount(
-    <LMap>
+  const root = render(
+    <LMap
+      bounds={[
+        [0, 0],
+        [1, 1],
+      ]}
+    >
       <WaypointsOverlay bounds={bounds} currentLevel={currentLevel} />
     </LMap>,
   );
 
-  expect(wrapper.find(WaypointMarker).exists()).toBeTruthy();
-  expect(wrapper.find(WaypointMarker).length).toBe(waypoints.length);
+  await waitFor(() => {
+    expect(root.queryAllByTestId('waypointMarker')).toBeTruthy();
```

Comment:
This a dangerous assertion because empty [] in some cases is true.

`if ([]) {console.log("It's true!")}; ` Run this in your browser's console.

So you may check .length > 0

HTH

Reply:
ohh okay, I overlooked this point! thanks for highlighting